### PR TITLE
refactor(oauth2): add current time injection point for testing

### LINKED ISSUE #8288: Add functor for updated time in `RefreshingCredentialsWrapper` in lieu of now().
Calling `std::chrono::system_time::now()` may lead to brittle unit tests. Consider adding an additional functor in its place.

### PR DESCRIPTION
**Pull request by scotthart**
fixes #8288

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8445)

### CODE REVIEW

File: google/cloud/internal/oauth2_refreshing_credentials_wrapper.cc [hunk 1]
```diff
@@ -20,6 +20,10 @@ namespace cloud {
 namespace oauth2_internal {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
+RefreshingCredentialsWrapper::RefreshingCredentialsWrapper(
+    CurrentTimeFn current_time_fn)
+    : current_time_fn_(std::move(current_time_fn)) {}
+
```
Comment (coryan) [inline google/cloud/internal/oauth2_refreshing_credentials_wrapper.cc:26]: Thanks, it seems that the `now` parameter becomes redundant?  Your call.

**Review by coryan — APPROVED** [review]

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8445)* status: 0 of 6 files reviewed, 1 unresolved discussion (waiting on @coryan)

___
*[google/cloud/internal/oauth2_refreshing_credentials_wrapper.cc, line 26 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8445#-Mwcyc-FfEHbO0ckhePn-r1-26:-MwhTAM91oR8kCEQBI_N:bmit84i) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/f7d08a2419ca5e151182610cd53d95e931ee59b2/google/cloud/internal/oauth2_refreshing_credentials_wrapper.cc#L26)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Thanks, it seems that the `now` parameter becomes redundant?  Your call.
</blockquote></details>

Removed the now parameter and just calling current_time_fn where needed.

**Review by coryan — APPROVED** [review]
> Reviewed 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8445)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @scotthart)

**Review by coryan — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8445#-:-MwiyiIl7fmfMKxD7DHL:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @scotthart)

**Pull request merged**